experimental/wgit/weigit_api.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import os
+import pathlib
+import sys
+
+import pygit2
+
+
+class WeiGit:
+    def __init__(self) -> None:
+        """
+        Planned Features:
+            1. create the wgit directory. Error, if already dir exists.
+            2. SHA1Store.init()
+            3. Create SHA1 .wgit/sha1_ref_count.json
+            3. Initialize a .git directory within the .wgit using `git init`.
+            4. add a .gitignore within the .wgit directory, so that the git repo within will ignore `sha1_ref_count.json`
+        """
+
+        # Make .wgit directory
+        try:
+            os.mkdir(".wgit")
+        except FileExistsError:
+            sys.stderr.write("WeiGit has been already Initialized \n")
+            sys.exit(0)
+
+        # if no .wgit dir then initialize the following
+        SHA1_store()
+
+        # create sha1_ref_count
+        with open(".wgit/sha1_ref_count.json", "w") as f:
+            print("The sha1_ref_count.json file has been created!")
```

Comment:
this is for debugging, right?

Reply:
yes, kept it for the skeleton. Probably better to remove it now!